pkg/daemon/update_test.go

```diff
@@ -77,7 +78,7 @@ func TestReconcilable(t *testing.T) {
 		Spec: mcfgv1.MachineConfigSpec{
 			Config: ignv2_2types.Config{
 				Ignition: ignv2_2types.Ignition{
-					Version: "0.0",
+					Version: "2.0.0",
```

Comment:
Is this supposed to be `2.2.0`?

Reply:
this is testing our general reconcilable logic. we don't reconcile ignition versions so this test is first checking that a change in ign versions fails (the old config and new config have diff ign versions), then the next part makes them both the same which is acceptable.

Reply:
those types of checks in practice come after the general ign validation that this pr adds. to make sure any machineconfig updates only make changes to the ign fields that we allow.